src/darwin/Framework/CHIP/MTROTAProviderDelegate.h

```diff
@@ -105,12 +102,13 @@ typedef void (^MTRApplyUpdateRequestCompletionHandlerDeprecated)(
                                     params:(MTROTASoftwareUpdateProviderClusterNotifyUpdateAppliedParams *)params
                                 completion:(MTRStatusCompletion)completion
     API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4));
+
 - (void)handleNotifyUpdateAppliedForNodeID:(NSNumber *)nodeID
                                 controller:(MTRDeviceController *)controller
                                     params:(MTROtaSoftwareUpdateProviderClusterNotifyUpdateAppliedParams *)params
                          completionHandler:(StatusCompletion)completionHandler
-    API_DEPRECATED(
-        "Please use the selector ending in completion:", ios(16.1, 16.4), macos(13.0, 13.3), watchos(9.1, 9.4), tvos(16.1, 16.4));
+    MTR_DEPRECATED_WITH_REPLACEMENT("handleNotifyUpdateAppliedForNodeID:controller:params:completion:", ios(16.1, 16.4),
```

Comment:
Question about the semantics of MTR_DEPRECATED_WITH_REPLACEMENT: The other selector has a different type for the `params` argument.  Does that still make it a WITH_REPLACEMENT case?

Reply:
I think in this case it's fine because `MTROtaSoftwareUpdateProviderClusterNotifyUpdateAppliedParams` is also meant to be replaced by the corresponding `MTROTA...` class.